File: lib/dto/user_dto.dart
// user_dto.dart
class UserDTO {
  final String username;
  final String email;
  final String realname;
  final String password;
  final String gender; 


  UserDTO({
    required this.username,
    required this.email,
    required this.realname,
    required this.password,
    required this.gender,
  });

  // DTO를 JSON으로 변환하는 메서드
  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'email': email,
      'realname': realname,
      'password': password,
      'gender': gender, // 성별 필드 포함
    };
  }
}

File: src/main/java/com/example/demo/DTO/UserDTO.java
package com.example.demo.DTO;

import com.example.demo.entity.User;

public class UserDTO {
    private long id; // 추가된 id 필드
    private String username;
    private String password;
    private String email;
    private String realname;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getRealname() {
        return realname;
    }

    public void setRealname(String realname) {
        this.realname = realname;
    }


    // 데이터를 엔티티로 변환하는 메서드
    public User toEntity() {
        User user = new User();
        user.setUsername(this.username);
        user.setPassword(this.password);
        user.setEmail(this.email);
        user.setRealname(this.realname);
        return user;
    }

    // 엔티티를 DTO로 변환하는 메서드
    public static UserDTO fromEntity(User user) {
        UserDTO dto = new UserDTO();
        dto.setId(user.getId()); // id 설정
        dto.setUsername(user.getUsername());
        dto.setPassword(user.getPassword());
        dto.setEmail(user.getEmail());
        dto.setRealname(user.getRealname());
        return dto;
    }
}
